Add Readme for Framework (#27918)
dotnet/aspnetcore: src/Framework/README.md

# Framework

> Contains packages that provide reference assembles, default sets APIs for building ASP.NET Core applications, and assets used for self-contained deployments.

Reviewer: I'm not sure what "default sets APIs" means. Please clarify. Please at least mention the shared framework and targeting pack installers that build on top of what's in this folder.

Further this doesn't make it clear the packages produced here exist _only_ to bundle assets created elsewhere and to add metadata.

Author: I'm not either, I copied these notes from the comments in the csproj for the assembly.  What's the correct summary?  Shared framework and targeting pack installers only?

Reviewer: ```suggestion
Contains packages that bundle reference and implementation assembles for Microsoft.AspNetCore.App.Runtime shared framework and for use in targeting pack installers.
```
Of course, this can probably be improved 😺

## Description

The following contains a description of each sub-directory in the `DataProtection` directory.

> - `App.Ref.Internal`: Contains internal reference assemblies, documentation, and other design-time assets

Reviewer 2: This package doesn't actually contain anything, it's sole purpose is to create a package identifying the non-stable version of AspNetCore at the time we last build the targeting pack

Author: Thanks updated, is that better?

> - `App.Ref.Internal`: Empty project for identifying the non-stable version of AspNetCore at the time the targeting pack was last built

Reviewer: May be worth mentioning the created package doesn't ship externally and shouldn't be important to 99.72% of developers

> - `App.Ref.Internal`: Empty project for identifying the non-stable version of AspNetCore at the time the targeting pack was  last built

Reviewer 2: nit - extra space between "was" and "last"

> - `App.Ref.Internal`: Empty project for identifying the non-stable version of AspNetCore at the time the targeting pack was last built. This package does not ship internally.

Reviewer: ```suggestion
- `App.Ref.Internal`: Empty project for identifying the non-stable version of AspNetCore at the time the targeting pack was last built. This package does not ship and is for internal use.
```

- `App.Ref`: Contains reference assemblies, documentation, and other design-time assets
- `App.Runtime`: Provides a default set of APIs for building ASP.NET Core applications, and assets used for self-contained deployments.

## Development Setup

### Build

> To build this specific project from source, you can follow the instructions [on building a subset of the code](https://github.com/dotnet/aspnetcore/blob/master/docs/BuildFromSource.md#building-a-subset-of-the-code).

Reviewer 3: ```suggestion
To build this specific project from source, follow the instructions [on building a subset of the code](https://github.com/dotnet/aspnetcore/blob/master/docs/BuildFromSource.md#building-a-subset-of-the-code).
```

> To build this specific project from source,  follow the instructions [on building a subset of the code](../../BuildFromSource.md#building-a-subset-of-the-code).

Reviewer: These two links to BuildFromSource.md are off a bit e.g.
```suggestion
To build this specific project from source,  follow the instructions [on building a subset of the code](../../docs/BuildFromSource.md#building-a-subset-of-the-code).
```

Reviewer: @captainsafia FYI I updated #27561 with this fix

### Test

> To run the tests for this project, [run the tests on the command line](../../BuildFromSource.md#running-tests-on-command-line) in this directory.

Reviewer: I left out a duplicate suggestion
```suggestion
To run the tests for this project, [run the tests on the command line](../../docs/BuildFromSource.md#running-tests-on-command-line) in this directory.
```

> To run the tests for this project, you can [run the tests on the command line](https://github.com/dotnet/aspnetcore/blob/master/docs/BuildFromSource.md#running-tests-on-command-line) in this directory.

Reviewer 3: ```suggestion
To run the tests for this project, [run the tests on the command line](https://github.com/dotnet/aspnetcore/blob/master/docs/BuildFromSource.md#running-tests-on-command-line) in this directory.
```

## More Information

> For more information, see the [ASP.NET Core README](../../README.md).

Reviewer: Why use a relative link here and HTTPS for the two links above❔ Not sure which is better but I'd rather we were consistent.

Author: I think this was an issue with the template that's been fixed after i copied, I'll resync
